scipy/fft/czt.py

```diff
@@ -0,0 +1,548 @@
+# This program is public domain
+# Authors: Paul Kienzle, Nadav Horesh
+"""
+Chirp z-transform.
+
+We provide two interfaces to the chirp z-transform: an object interface
+which precalculates part of the transform and can be applied efficiently
+to many different data sets, and a functional interface which is applied
+only to the given data set.
+
+Transforms
+----------
+
+CZT : callable (x, axis=-1) -> array
+   Define a chirp z-transform that can be applied to different signals.
+ZoomFFT : callable (x, axis=-1) -> array
+   Define a Fourier transform on a range of frequencies.
+
+Functions
+---------
+
+czt : array
+   Compute the chirp z-transform for a signal.
+zoomfft : array
+   Compute the Fourier transform on a range of frequencies.
+"""
+
+import cmath
+import numpy as np
+from numpy import pi, arange
+from scipy.fft import fft, ifft, next_fast_len
+
+__all__ = ['czt', 'zoomfft', 'CZT', 'ZoomFFT', 'czt_points']
+
+
+def _validate_sizes(n, m):
+    if n < 1:
+        raise ValueError("Invalid number of CZT data "
+                         "points (%d) specified." % n)
+
+    if int(n) != n:
+        raise ValueError('n must be a positive integer')
+
+    if m is None:
+        m = n
+    elif m < 1:
+        raise ValueError("Invalid number of CZT output "
+                         "points (%d) specified." % n)
+    elif int(m) != m:
+        raise ValueError('m must be a positive integer')
+
+    return m
+
+
+def czt_points(m, w=None, a=1+0j):
+    """
+    Return the points at which the chirp z-transform is computed.
+
+    Parameters
+    ----------
+    m : int, optional
+        The number of points desired.
+    w : complex, optional
+        The ratio between points in each step.
+    a : complex, optional
+        The starting point in the complex plane.  Default is 1+0j.
+
+    Returns
+    -------
+    out : ndarray
+        The points in the Z plane at which `CZT` samples the z-transform,
+        when called with arguments `m`, `w`, and `a`, as complex numbers.
+
+    See Also
+    --------
+    CZT : Class that creates a callable chirp z-transform function.
+    czt : Convenience function for quickly calculating CZT.
+
+    Examples
+    --------
+    Plot the points of a 16-point FFT:
+
+    >>> from scipy.fft import czt_points
+    >>> points = czt_points(16)
+    >>> import matplotlib.pyplot as plt
+    >>> plt.plot(points.real, points.imag, 'o')
+    >>> plt.gca().add_patch(plt.Circle((0,0), radius=1, fill=False, alpha=.3))
+    >>> plt.axis('equal')
+    >>> plt.show()
+
+    and a 91-point logarithmic spiral that crosses the unit circle:
+
+    >>> m, w, a = 91, 0.995*np.exp(-1j*np.pi*.05), 0.8*np.exp(1j*np.pi/6)
+    >>> points = czt_points(m, w, a)
+    >>> plt.plot(points.real, points.imag, 'o')
+    >>> plt.gca().add_patch(plt.Circle((0,0), radius=1, fill=False, alpha=.3))
+    >>> plt.axis('equal')
+    >>> plt.show()
+    """
+    m = _validate_sizes(1, m)
```

Comment:
Especially this one

Reply:
Oh, I see, the operator.index() change modifies the variables instead of just checking them. 

It removes the helpful error messages, too.  Instead of `m must be a positive integer`, it says `'float' object cannot be interpreted as an integer`.  

If they are integers already then they shouldn't need to be modified?  It will convert things like `numpy.int32` to `int`, but that isn't necessary.

Reply:
Should it just check that the type is integer instead of checking that the value is integer, but otherwise keep the same functionality of helpful error message and not modifying things that don't need to be modified?

Reply:
I think it would be best to stick with the agreed-upon SciPy convention of using `operator.index` in this PR. Then if we want to change the integer-checking behavior in SciPy, we talk about it in a separate issue and change all instances at a later date